core/kogitoinfra/knative_reconciler.go

```diff
@@ -12,54 +12,49 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package controllers
+package kogitoinfra
 
 import (
 	"fmt"
-
-	"github.com/kiegroup/kogito-cloud-operator/pkg/client/kubernetes"
-	"github.com/kiegroup/kogito-cloud-operator/pkg/infrastructure"
-	"github.com/kiegroup/kogito-cloud-operator/pkg/logger"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	eventingv1 "knative.dev/eventing/pkg/apis/eventing/v1"
+	"github.com/kiegroup/kogito-cloud-operator/core/infrastructure"
+	"k8s.io/apimachinery/pkg/types"
 )
 
-func initknativeInfraReconciler(context targetContext) *knativeInfraReconciler {
-	log := logger.GetLogger("knative")
+func initknativeInfraReconciler(context targetContext) Reconciler {
+	context.log = context.log.WithValues("resource", "knative")
 	return &knativeInfraReconciler{
 		targetContext: context,
-		log:           log,
 	}
 }
 
 // knativeInfraReconciler for Knative resources reconciliation
 type knativeInfraReconciler struct {
 	targetContext
-	log logger.Logger
 }
 
 // Reconcile ...
 func (k *knativeInfraReconciler) Reconcile() (requeue bool, resultErr error) {
-
-	if !infrastructure.IsKnativeEventingAvailable(k.client) {
-		return false, errorForResourceAPINotFound(&k.instance.Spec.Resource)
+	knativeHandler := infrastructure.NewKnativeHandler(k.client)
+	if !knativeHandler.IsKnativeEventingAvailable() {
+		return false, errorForResourceAPINotFound(k.instance.GetSpec().GetResource().APIVersion)
 	}
 
-	if len(k.instance.Spec.Resource.Name) > 0 {
-		ns := k.instance.Spec.Resource.Namespace
+	if len(k.instance.GetSpec().GetResource().Name) > 0 {
+		ns := k.instance.GetSpec().GetResource().Namespace
 		if len(ns) == 0 {
-			k.log.Debug("Namespace not defined, fetching from the current", "Namespace", k.instance.Namespace)
-			ns = k.instance.Namespace
+			k.log.Debug("Namespace not defined, setting to current namespace")
+			ns = k.instance.GetNamespace()
 		}
-		broker := eventingv1.Broker{ObjectMeta: metav1.ObjectMeta{Namespace: ns, Name: k.instance.Spec.Resource.Name}}
-		if exists, resultErr := kubernetes.ResourceC(k.client).Fetch(&broker); resultErr != nil {
+
+		broker, resultErr := knativeHandler.CreateBroker(types.NamespacedName{Name: k.instance.GetSpec().GetResource().Name, Namespace: ns})
```

Comment:
In current version the broker creation is left on user. Is there any specific reason to change it to create the broker by operator?

Reply:
functionality is same as before. `CreateBroker` function name is wrong. Internally its only fetching existing broker. I had rename this function to `FetchBroker`.